xbmc/utils/DumbBufferObject.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DumbBufferObject.h"
+
+#include "ServiceBroker.h"
+#include "utils/BufferObjectFactory.h"
+#include "utils/log.h"
+#include "windowing/gbm/WinSystemGbm.h"
+#include "windowing/gbm/WinSystemGbmEGLContext.h"
+
+#include <drm/drm_fourcc.h>
+#include <sys/mman.h>
+
+using namespace KODI::WINDOWING::GBM;
+
+std::unique_ptr<CBufferObject> CDumbBufferObject::Create()
+{
+  return std::make_unique<CDumbBufferObject>();
+}
+
+void CDumbBufferObject::Register()
+{
+  CBufferObjectFactory::RegisterBufferObject(CDumbBufferObject::Create);
+}
+
+CDumbBufferObject::CDumbBufferObject()
+{
+  m_device = static_cast<CWinSystemGbmEGLContext*>(CServiceBroker::GetWinSystem())
+                 ->GetDrm()
+                 ->GetFileDescriptor();
```

Comment:
How about breaking this into multiple lines?

```c++
CWinSystemGbmEGLContext* winSystem = static_cast<CWinSystemGbmEGLContext*>(CServiceBroker::GetWinSystem());

m_device = winSystem->GetDrm()->GetFileDescriptor();
```

As it is, because they occupy the same line, you can say that m_device has a dependency on `CServiceBroker`. Moving the call to `CServiceBroker` to a separate line breaks this dependency. I personally push all dependencies on `CServiceBroker` to the top of the function, as far away from internal details as I can.

Reply:
Updated, thanks for the input